app/services/utils/raster.py

```diff
@@ -1,23 +1,46 @@
+import base64
+
 from rio_tiler.io.rasterio import Reader
 from rasterio import features
 from geopandas import GeoDataFrame
 from shapely import geometry
 from typing import Any
 
 
+from app.middleware.log_middleware import logger
+
+
 # TODO: become generic in order to be able to reuse
 def crop_raster(raster_path, polygon):
     with Reader(input=raster_path, options={}) as image:
         img = image.feature(polygon)
 
-    return img.render(
-        add_mask=True,
-        colormap={
-            0: (255, 0, 0, 255),
-            1: (128, 204, 102, 255),
-            2: (232, 214, 107, 255),
-        },
-    )
+    colormap = {
+        0: (255, 0, 0, 255),
+        1: (128, 204, 102, 255),
+        2: (232, 214, 107, 255),
+    }
+
+    base64_images = {}
+
+    for category, color in colormap.items():
+        try:
```

Comment:
I suggest to not capture the exception here and let it rise to the exception handler. But if you wish to send a custom message to the log and to the user, don't log on this file and raise a ServerError as is being done in the collection file

Reply:
ans in the second case, let the messages in english